<a href="https://colab.research.google.com/github/hiramaral/CN_UEA_2022/blob/main/Questao2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
import math


def function(x):
    return (9.8*x/13)*(1 - math.e**(-13*6/x)) - 31


def bissection(a, b):
    if function(a)*function(b) >= 0:
        print("Intervalo Inválido!")
        return
    
    c = 0
    k = 1

    while(abs(function((a + b)/2)) >= 0.0001):
        c = (a + b)/2

        if function(c) == 0.0:
            print("Iteracao: "+str(k))
            print("Intervalo: ["+str(a)+", "+str(b)+"], valor do x médio: "+str(c))
            break

 
        if (function(a)*function(c) < 0):
            b = c
            print("Iteracao: "+str(k))
            print("Intervalo: ["+str(a)+", "+str(b)+"], valor do x médio: "+str(c))
        else:
            a = c
            print("Iteracao: "+str(k))
            print("Intervalo: ["+str(a)+", "+str(b)+"], valor do x médio: "+str(c))
        k += 1

In [ ]:
bissection(52, 55) 

Iteracao: 1
Intervalo: [53.5, 55], valor do x médio: 53.5
Iteracao: 2
Intervalo: [53.5, 54.25], valor do x médio: 54.25
Iteracao: 3
Intervalo: [53.5, 53.875], valor do x médio: 53.875
Iteracao: 4
Intervalo: [53.5, 53.6875], valor do x médio: 53.6875
Iteracao: 5
Intervalo: [53.59375, 53.6875], valor do x médio: 53.59375
Iteracao: 6
Intervalo: [53.640625, 53.6875], valor do x médio: 53.640625
Iteracao: 7
Intervalo: [53.6640625, 53.6875], valor do x médio: 53.6640625
Iteracao: 8
Intervalo: [53.6640625, 53.67578125], valor do x médio: 53.67578125


Iteracao: 2


Intervalo: [53.5, 54.25], valor do x médio: 54.25


Iteracao: 3


Intervalo: [53.5, 53.875], valor do x médio: 53.875


Iteracao: 4


Intervalo: [53.5, 53.6875], valor do x médio: 53.6875